src/dmd/dclass.d

```diff
@@ -748,6 +749,24 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
                 if (fd == fdmatch)
                     continue;
 
+                /* Functions overriding interface functions for extern(C++) with VC++
+                 * are not in the normal vtbl, but in vtblFinal. If the implementation
+                 * is again overridden in a child class, both would be found here.
+                 * The function in the child class should override the function
+                 * in the base class, which is done here, because searchVtbl is first
+                 * called for the child class. Checking seenInterfaceVirtual makes
+                 * sure, that the compared functions are not in the same vtbl.
+                 */
+                if (fd.interfaceVirtual &&
```

Comment:
While the result looks good, I wonder whether the explicit comparison of the type modifier is necessary (should be part of `Type.equals` AFAICT), but also why only an exact match is considered and not a covariant overload, too

> Checking seenInterfaceVirtual makes sure, that the compared functions are not in the same vtbl.

This seems a bit confusing as `seenInterfaceVirtual` is reset for each `vtbl`.

Reply:
You are right, that it should also allow covariant functions. Currently they would still get the error. I will change that and extend the test.

The function searchVtbl is called for the current class and every base class. If multiple matching functions are found in a single call of searchVtbl, then seenInterfaceVirtual will be true for the second and the error message will be generated.